### PR TITLE
Unix socket support

### PR DESCRIPTION
**Pull request by BadSingleton**

### CODE REVIEW

File: rpyc/core/stream.py [hunk 1]
```diff
@@ -7,7 +7,14 @@
...
 from rpyc.lib import safe_import, Timeout, socket_backoff_connect
-from rpyc.lib.compat import poll, select_error, BYTES_LITERAL, get_exc_errno, maxint  # noqa: F401
+from rpyc.lib.compat import (
+    poll,
+    select_error,
+    BYTES_LITERAL,
+    get_exc_errno,
+    maxint,
+    unix_socket,
+)
```
Comment (benoithudson) [inline rpyc/core/stream.py:17]: what happened to `# noqa: F401` (what does it mean anyway?)

File: rpyc/lib/win_unix_socket.py [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+MAX_PATH_LENGTH = 260
+WSAEWOULDBLOCK = 10035
+SOL_SOCKET = 65535
+SO_ERROR = 4103
+EISCONN = 10056
+WSA_INVALID_PARAMETER = 87
+FORMAT_MESSAGE_ALLOCATE_BUFFER = 0x100
+FORMAT_MESSAGE_FROM_SYSTEM = 0x1000
+FORMAT_MESSAGE_IGNORE_INSERTS = 0x200
+FORMAT_MESSAGE_FLAGS = (FORMAT_MESSAGE_ALLOCATE_BUFFER
```
Comment (benoithudson) [inline rpyc/lib/win_unix_socket.py]: Can you PEP-8 lint this? I think this line is wrong but I'm not sure.

File: rpyc/lib/win_unix_socket.py [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+
+        Returns None. After this call, this socket will be connected to the server
+        unless an exception was raised.
+        """
+        path = bytes(path)
+        # copy the name including the null terminator (cpython has null-terminated strings under the hood)
+        nbytes = len(path) + 1
+        if nbytes > MAX_PATH_LENGTH:
+            raise socket.error(ERROR_BAD_PATHNAME,
+                    "Socket path name too long ({} bytes, max {})".format(nbytes, MAX_PATH_LENGTH)
```
Comment (benoithudson) [inline rpyc/lib/win_unix_socket.py]: I remember you refactored this to avoid the duplication; are we working from the correct branch?

**Review by benoithudson — COMMENTED** [review]
Looks like you missed a commit somewhere, with the refactoring.

File: rpyc/lib/win_unix_socket.py (hunk 2 above)
Comment (BadSingleton) [reply rpyc/lib/win_unix_socket.py]: I will

File: rpyc/core/stream.py (hunk 1 above)
Comment (BadSingleton) [reply rpyc/core/stream.py:17]: lost in patching process. I'll put it back. It's a hint to some linters not to flag the line

Comment (BadSingleton) [reply rpyc/core/stream.py:17]: `noqa: F401` is for no warning on unused imports

**Comment by BadSingleton** [comment]
I went ahead and linted the whole thing. Found a few issues in win_unix_socket; mainly undefined names. We'll have to backport this

File: rpyc/lib/win_unix_socket.py [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+                    # and waits for the async IO to finish.
+                    break
+                else:
+                    # if we're here, the error is scary
+                    raise_socket(err)
+
+
+def _make_address(path: bytes):
+    """
+    Creates the sockaddr_un srtucture based on the path given to this method.
```
Comment (benoithudson) [inline rpyc/lib/win_unix_socket.py]: srtucture => structure

File: rpyc/lib/win_unix_socket.py [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+                elif timeout > 0 and err == WSAEWOULDBLOCK:
+                    # break this loop, continue outer loop that does a poll()
+                    # and waits for the async IO to finish.
+                    break
+                else:
+                    # if we're here, the error is scary
+                    raise_socket(err)
+
+
+def _make_address(path: bytes):
```
Comment (benoithudson) [inline rpyc/lib/win_unix_socket.py]: does this parse in py 2.7?

**Comment by benoithudson** [comment]
We need to make sure this runs in python 2.7 as well. Some of the string stuff seems to be py3 only.

File: rpyc/lib/win_unix_socket.py (hunk 5 above)
Comment (BadSingleton) [reply rpyc/lib/win_unix_socket.py]: It doesn't. I fixed this as you were reviewing

**Review by benoithudson — APPROVED** [review]

**Pull request merged**